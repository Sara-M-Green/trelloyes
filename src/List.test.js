import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ListClass from './List';
import CardsClass from './Card';

//test suite
describe('ListClass Componenet', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ListClass key='1' header='First list' cards={[ 'a', 'b', 'e', 'f', 'g', 'j', 'l', 'm' ]} />, div)
        ReactDOM.unmountComponentAtNode(div);
    });

    //snapshot test
    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<ListClass key='2' header='Second list' cards={['b', 'c', 'd', 'f', 'h', 'i', 'k']} />)
            .toJSON();
        expect(tree).toMatchSnapshot()
    });
})