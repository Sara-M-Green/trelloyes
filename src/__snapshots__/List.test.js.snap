// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListClass Componenet renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Second list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3
        className="Card-title"
      />
      <p
        className="Card-content"
      />
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3
        className="Card-title"
      />
      <p
        className="Card-content"
      />
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3
        className="Card-title"
      />
      <p
        className="Card-content"
      />
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3
        className="Card-title"
      />
      <p
        className="Card-content"
      />
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3
        className="Card-title"
      />
      <p
        className="Card-content"
      />
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3
        className="Card-title"
      />
      <p
        className="Card-content"
      />
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3
        className="Card-title"
      />
      <p
        className="Card-content"
      />
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
